ark-cli/src/commands/link/utils.rs

```diff
@@ -38,6 +38,7 @@ pub fn load_link(
                 Err(AppError::LinkLoadError(format!(
                     "Path {:?} was requested. But id {} maps to path {:?}",
                     path,
+                    // FIXME: If `load_link` needs id, then it shouldn't be an optional parameter
                     id.clone().unwrap(),
                     path2,
                 )))
```

Comment:
An interesting question is how we can define app-specific resource kinds and commands related to these specific resource kinds, handling them uniformly?

That could be a plugin system. So, any app developer would create not only wrappers for each platform, but also CLI commands related to resource kinds this developer introduces. E.g. `.link` is custom file format defined by us and it contains `Link to Web` resource. It consists only of URL, but we also define some metadata and possible properties for it, e.g. domain and title.

Reply:
Interesting idea. I created issue #69 to track this proposal. We might need more time to discuss and define a plan before starting implementation.

Reply:
Let's remove the `FIXME` before merging. We have generic #69 but specific link-related issue could be created, too.